lib/r10k/git.rb

```diff
@@ -19,8 +20,12 @@ module Git
           :feature => :shellgit,
           :module  => R10K::Git::ShellGit,
         }
-      ],
-      [ :rugged,
+      ]
+    ]
+
+    unless R10K::Util::Platform.jruby?
+      @providers << [
```

Comment:
It must have been one of the now-skipped test scenarios that was causing the on-set proc to be evaluated (which blows up if the Rugged module isn't defined)...